girder-dandi-archive/girder_dandi_archive/util.py

```diff
@@ -0,0 +1,38 @@
+import re
+
+from girder.models.collection import Collection
+from girder.utility import setting_utilities
+from girder.exceptions import ValidationException
+
+DANDISET_ID_COUNTER = "dandi.id_counter"
+DANDISET_ID_LENGTH = 6
+DANDI_STAGING_COLLECTION_NAME = "Dandiset Staging"
+
+dandiset_id_pattern = r"^\d{6}$"
+
+
+@setting_utilities.validator(DANDISET_ID_COUNTER)
+def _validate(doc):
+    if len(str(doc["value"])) > DANDISET_ID_LENGTH:
+        raise ValidationException("Dandiset ID limit exceeded", "value")
+
+
+def validate_dandiset_id(dandiset_id):
+    return bool(re.match(dandiset_id_pattern, dandiset_id))
+
+
+def pad_dandiset_id(dandiset_id):
+    id_str = str(dandiset_id)
```

Comment:
You can do all of this with just:
```
return f"{dandiset_id:0{DANDISET_ID_LENGTH}d}"
```

Also, `pad_dandiset_id` probably doesn't need to be a utility function (i.e. just inline the f-string expression where it's used), since the IDs should always be handled as strings (except when they're created).